tests/tape/test_qnode.py

```diff
@@ -323,12 +323,52 @@ def func(x):
             qml.RX(x, wires=0)
             return qml.expval(qml.PauliZ(0))
 
-        qn = qml.QNode(func, dev)
+        qn = qml.QNode(func, dev, diff_method="finite-diff")
 
         assert (
             qn.__repr__()
-            == "<QNode: wires=1, device='default.qubit.autograd', interface='autograd', diff_method='best'>"
+            == "<QNode: wires=1, device='default.qubit', interface='autograd', diff_method='finite-diff'>"
         )
+        assert qn.diff_method_change == False
+
+    def test_qnode_best_diff_method(self, monkeypatch):
+        """Test that selected "best" diff_method is available after QNode initialization."""
+        dev = qml.device("default.qubit", wires=1)
+
+        def func(x):
+            qml.RX(x, wires=0)
+            return qml.expval(qml.PauliZ(0))
+
+        qn = qml.QNode(func, dev)
+
+        assert qn.diff_method == "backprop"
+        assert qn.diff_method_change
+
+        dev = qml.device("default.mixed", wires=1)
+        qn = qml.QNode(func, dev)
+
+        assert qn.diff_method == "parameter-shift"
+        assert qn.diff_method_change
+
+        dev = qml.device("default.qubit", wires=1)
+        monkeypatch.setitem(dev._capabilities, "passthru_interface", "some_interface")
+        monkeypatch.setitem(dev._capabilities, "provides_jacobian", True)
+
+        qn = qml.QNode(func, dev)
+        assert qn.diff_method == "device"
+        assert qn.diff_method_change
+
+        monkeypatch.setitem(dev._capabilities, "provides_jacobian", False)
+
+        def capabilities(cls):
+            capabilities = cls._capabilities
+            capabilities.update(model="None")
+            return capabilities
+
+        monkeypatch.setattr(qml.devices.DefaultQubit, "capabilities", capabilities)
+        qn = qml.QNode(func, dev)
+        assert qn.diff_method == "finite-diff"
+        assert qn.diff_method_change

```

Comment:
Perhaps a few comments can be added in the tests explaining why certain things are being monkeypatched and what exactly is being tested.

Reply:
I have added a comment and made different tests for different possibilities of "best"